java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -3386,4 +3406,56 @@ void testORCWriteToFileUncompressed() throws IOException {
       tempFileUncompressed.delete();
     }
   }
+
+  @Test
+  void testStructColumnFilter() {
+    HostColumnVector.ColumnBuilder.StructType type = new HostColumnVector.ColumnBuilder.StructType(true, 4);
+    type.addChild(new HostColumnVector.ColumnBuilder.BasicType(true, 4, DType.INT32));
+    type.addChild(new HostColumnVector.ColumnBuilder.BasicType(true, 4, DType.INT64));
+    HostColumnVector.ColumnBuilder.StructType expectedType = new HostColumnVector.ColumnBuilder.StructType(true, 2);
+    expectedType.addChild(new HostColumnVector.ColumnBuilder.BasicType(true, 2, DType.INT32));
+    expectedType.addChild(new HostColumnVector.ColumnBuilder.BasicType(true, 2, DType.INT64));
```

Comment:
Wow this looks painful to write.  Can we please have a follow on issue to clean this up?  When building up test data we should just need to provide the data, and some hint of what the type of the data should be.

Reply:
I will start a follow on if I cant fix it here. I know this is not great :(